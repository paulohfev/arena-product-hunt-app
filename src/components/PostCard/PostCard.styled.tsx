import Image from 'next/image'

import styled from 'styled-components'

export const PostCardWrapper = styled.div`
  background-color: #ffffff;
  border-radius: 8px;
  cursor: pointer;
  display: flex;
  gap: 16px;
  margin-bottom: 1rem;
  padding: 16px;
  width: 100%;

  &:hover {
    transform: translateY(-2px);
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  }
`

export const PostBody = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
`

export const PostTextContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;
`

export const PostImage = styled(Image)`
  border-radius: 16px;
`

export const PostTitle = styled.h4`
  font-size: 18px;
  font-weight: 500;
`

export const PostTagline = styled.p`
  color: #666;
`

export const PostStats = styled.div`
  align-items: center;
  border: 1px solid #e0e0e0;
  border-radius: 10px;
  cursor: pointer;
  display: flex;
  flex-direction: column;
  font-size: 12px;
  gap: 4px;
  height: 48px;
  justify-content: center;
  width: 48px;
`
