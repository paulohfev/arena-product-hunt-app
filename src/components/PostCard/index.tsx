import Image from 'next/image'

import { Post } from '@/types/Post'

import {
  PostBody,
  PostCardWrapper,
  PostImage,
  PostStats,
  PostTagline,
  PostTextContainer,
  PostTitle,
} from './PostCard.styled'

export type PostCardProps = {
  post: Post
}

const PostCard: React.FC<PostCardProps> = ({ post }) => {
  return (
    <PostCardWrapper key={post.id}>
      <PostImage src={post.thumbnail.url} alt={post.name} width={48} height={48} />

      <PostBody>
        <PostTextContainer>
          <PostTitle>{post.name}</PostTitle>
          <PostTagline>{post.tagline}</PostTagline>
        </PostTextContainer>
        <PostStats>
          <Image src='/icons/triangle.png' alt='vote' width={12} height={12} />

          {post.votesCount}
        </PostStats>
      </PostBody>
    </PostCardWrapper>
  )
}

export default PostCard
