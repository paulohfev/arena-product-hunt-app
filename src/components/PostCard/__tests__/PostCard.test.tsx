import { render } from '@testing-library/react'

import PostCard from '../index'

const mockPost = {
  id: '1',
  name: 'Test Post',
  tagline: 'Test Tagline',
  description: 'Test Description',
  votesCount: 100,
  url: 'https://example.com',
  thumbnail: {
    url: 'https://example.com/image.jpg',
  },
}

describe('PostCard', () => {
  it('matches snapshot', () => {
    const { container } = render(<PostCard post={mockPost} />)
    expect(container).toMatchSnapshot()
  })
})
