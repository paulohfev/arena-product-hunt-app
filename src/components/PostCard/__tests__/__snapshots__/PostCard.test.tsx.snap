// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard matches snapshot 1`] = `
<div>
  <a
    class="sc-689d80bd-0 cqta-Dy"
    href="/posts/1"
  >
    <div
      class="sc-689d80bd-1 gihtda"
    >
      <img
        alt="Test Post"
        class="sc-689d80bd-4 jbEYvH"
        data-nimg="1"
        decoding="async"
        height="48"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=96&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="48"
      />
      <div
        class="sc-689d80bd-2 cNTLXF"
      >
        <div
          class="sc-689d80bd-3 jvqCyp"
        >
          <h4
            class="sc-689d80bd-5 jrPwKY"
          >
            Test Post
          </h4>
          <p
            class="sc-689d80bd-6 ctLJBT"
          >
            Test Tagline
          </p>
        </div>
        <div
          class="sc-689d80bd-7 dnAaWe"
        >
          <img
            alt="vote"
            data-nimg="1"
            decoding="async"
            height="12"
            loading="lazy"
            src="/_next/image?url=%2Ficons%2Ftriangle.png&w=32&q=75"
            srcset="/_next/image?url=%2Ficons%2Ftriangle.png&w=16&q=75 1x, /_next/image?url=%2Ficons%2Ftriangle.png&w=32&q=75 2x"
            style="color: transparent;"
            width="12"
          />
          100
        </div>
      </div>
    </div>
  </a>
</div>
`;
